components/nuki_lock/nuki_lock.cpp

```diff
@@ -533,6 +690,219 @@ void NukiLockComponent::set_pairing_mode(bool enabled) {
     }
 }
 
+void NukiLockComponent::set_auto_unlatch_enabled(bool enabled) {
+
+    Nuki::CmdResult cmdResult = this->nukiLock_.enableAutoUnlatch(enabled);
+
+    #ifdef USE_SWITCH
```

Comment:
I think it makes sense to just export that to a method instead of having so many repetitions.

Reply:
Do you have an example for me to understand it better?
 Something like this:
 ```c++
 void NukiLockComponent::set_config_switch(std::string config, bool value) {

    #ifdef USE_SWITCH
    Nuki::CmdResult cmdResult;

    // Update Config
    if(config == "auto_unlatch_enabled")
    {
        cmdResult = this->nuki_lock_.enableAutoUnlatch(value);
    }
    else if(config == "button_enabled")
    {
        cmdResult = this->nuki_lock_.enableButton(value);
    }

    if(cmdResult == Nuki::CmdResult::Success)
    {
        if(config == "auto_unlatch_enabled")
        {
            if (this->auto_unlatch_enabled_switch_ != nullptr) this->auto_unlatch_enabled_switch_->publish_state(enabled);
        }
        else if(config == "button_enabled")
        {
            if (this->button_enabled_switch_ != nullptr) this->button_enabled_switch_->publish_state(enabled);
        }        
        
        this->config_update_ = true;
    }
    #endif
}
 ```

Reply:
I could also just optimistically update the state of the entities and leave the publish_state out of this function